tests/test_default_extensions.py

```diff
@@ -25,8 +26,7 @@ def test_jinja2_time_extension(tmp_path):
     changelog_file = os.path.join(project_dir, 'HISTORY.rst')
     assert os.path.isfile(changelog_file)
 
-    with open(changelog_file, encoding='utf-8') as f:
-        changelog_lines = f.readlines()
+    changelog_lines = Path(changelog_file).open(encoding='utf-8').readlines()
```

Comment:
```suggestion
    with Path(changelog_file).open(encoding='utf-8') as f:
        changelog_lines = f.readlines()
```

Reply:
I leaved 'with' blocks where several operations done or f object checked itself after, for such simple cases I prefer oneliner.

Reply:
But as for handler you are right, chaining do not close it... good catch.